runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilitySession.java

```diff
@@ -53,8 +53,15 @@ void incrementLapCount(Client client, XpTrackerService xpTrackerService)
 		final int currentExp = client.getSkillExperience(Skill.AGILITY);
 		final int goalXp = xpTrackerService.getEndGoalXp(Skill.AGILITY);
 		final int goalRemainingXp = goalXp - currentExp;
+		double courseTotalExp = course.getTotalXp();
+		if (course == Courses.PYRAMID)
+		{
+			// agility pyramid has a bonus exp drop on the last obstacle that scales with player level and caps at 1000
+			// the bonus is not already accounted for in the total exp number in the courses enum
+			courseTotalExp += Math.min(course.getLastObstacleXp() + 8 * client.getRealSkillLevel(Skill.AGILITY), 1000);
```

Comment:
Based on my comment above, this isn't correct. Additionally, this would not be accurate for any goal level which is more than 1 level away and is below the bonus xp cap. (related: #10076)

Reply:
im not sure being level aware is something this can do, its miles more accurate that currently though.  If you have any suggestions on how to get it to be level aware i would welcome them.

Reply:
No suggestions as of now, but it would likely be good to add a `// TODO` comment or something of the sort.